### PR TITLE
Updates package manager versions

### PR DESCRIPTION
**Pull request by arcanis**
Upgrade package manager versions to their latest releases.

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
       }
     },
     "pnpm": {
-      "default": "6.29.1",
+      "default": "7.1.0",
```
Comment (arcanis) [inline config.json:31]: Ping @zkochan @aduh95, I'm not sure whether we can/should/must upgrade the default pnpm from 6 to 7, what do you think (I'm worried it would make it a breaking change for Node)?

Also cc @mcollina: with your proposal (automatically retrieving the latest versions), in a situation like this one, do you think Corepack should just upgrade from 6 to 7?

Comment (aduh95) [reply config.json:31]: I think we should aim for the latest, Corepack is still experimental in Node.js so breaking changes are still expected (and the upside is worth the potential breakage imo).

I think Matteo proposal was we should use a semver-range as the default (so `7.x` in this instance), so on a fresh install of Node.js you're less likely to get a version that contains known security issues.

Comment (zkochan) [reply config.json:31]: I don't know, whatever the Node.js team thinks is appropriate.
But do update pnpm to v7 in Node.js 18.

Comment (ljharb) [reply config.json:31]: It should use ^ by default, meaning that updating a major should be considered a breaking change. Exceptions should also be possible, like for npm 8, where the only breaking change wrt 7 was dropping node 10.

**Review by aduh95 — APPROVED** [review]
LGTM

File: config.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "definitions": {
     "npm": {
-      "default": "8.4.0",
+      "default": "8.10.0",
```
Comment (styfle) [inline config.json:4]: Forgive my drive-by comment here, but I thought npm wasn't supported with corepack?

Comment (merceyz) [reply config.json:4]: It has always been supported just not enabled by default when you run `corepack enable`, you can explicitly enable it with `corepack enable npm`
Ref https://github.com/nodejs/corepack/pull/24

> Note that the npm shims will not be installed unless explicitly requested, as npm is currently distributed with Node.js through other means.
https://github.com/nodejs/corepack/blob/648bd95e947d0a6be0beba4c594de127d94da7d9/README.md#corepack-enable--name
